### PR TITLE
Vestax VCI 400 controller updates

### PR DESCRIPTION
**Pull request by ywwg**
Add brake stop and spinback stop as shift modes for play and censor button.  Also fix out of date library browsing binding.  The diff for the xml file is horrible, but that should go away in the future with PR #1856

### CODE REVIEW

**Comment by Be-ing** [comment]
Thanks for taking this up. Let's target it at the 2.2 branch. Even if we don't get it all done by the 2.2.0 release, this could be included in a 2.2.x release.

**Comment by Be-ing** [comment]
Could you rebase your branch on 2.2?

**Comment by ywwg** [comment]
looks like I can't change the source branch, but I rebased it

**Comment by ywwg** [comment]
(I need to do one more check before we merge this in)

**Comment by Be-ing** [comment]
Were you intending to update the effects mapping for this controller in this PR? Or a future PR?

**Comment by ywwg** [comment]
I didn't know effects bindings had changed, do you have an example / doc of what's new?

**Comment by Be-ing** [comment]
All the old 2.0 ControlObjects still work, but this mapping hasn't been updated for the effects UI introduced in 2.1. The [EffectUnit in Components](https://mixxx.org/wiki/doku.php/components_js#effectunit) should work well with this controller.

File: res/controllers/common-controller-scripts.js [hunk 1]
```diff
@@ -326,9 +326,8 @@ script.midiPitch = function (LSB, MSB, status) {
    -------- ------------------------------------------------------ */
 script.spinback = function(channel, control, value, status, group, factor, rate) {
     // if brake is called without defined factor and rate, reset to defaults
-    if (factor === undefined && rate === undefined) {
```
Comment (Be-ing) [inline res/controllers/common-controller-scripts.js]: Why did you remove this?

It would be better to reimplement the `rate` default with a separate statement though. In the event that `factor` was defined but `rate` was not, both parameters would be set to the defaults.

**Review by Be-ing — CHANGES_REQUESTED** [review]
One small note about the code, otherwise LGTM. I updated the [wiki page](https://mixxx.org/wiki/doku.php/vestax_vci-400) and removed info about an obsolete mapping. Something I am wondering about this controller is if timecode control works with the RCA inputs. From what I saw searching the web, the RCA inputs don't have phono preamps, but I do see a gain control that I suppose might provide enough gain for vinyl control to work.

**Comment by Be-ing** [comment]
A few other questions about the hardware capabilities that would be nice to note on the wiki:
  * Are the microphone inputs available to the computer?
  * Can it run on USB bus power or does it require its own power supply?

**Comment by ywwg** [comment]
> All the old 2.0 ControlObjects still work, but this mapping hasn't been updated to make use of the effects UI introduced in 2.1. The [EffectUnit in Components](https://mixxx.org/wiki/doku.php/components_js#effectunit) should work well with this controller.

I took a look at the new effectunit component, and it will be more work than I expected to update the config to work with it. It's not just a trivial swap-out of the existing controls, and I will have to figure out where to relocate the quantize and keylock buttons.  I'd really prefer to do that in a different PR because this current PR fixes the controller being broken for 2.2.

**Comment by ywwg** [comment]
> * Are the microphone inputs available to the computer?
> * Can it run on USB bus power or does it require its own power supply?

Yes the mic inputs are available to the computer.  

No, it can't run on USB power.  In fact, the controller can work as a two deck hardware mixer without being connected to USB at all!

File: res/controllers/common-controller-scripts.js (hunk 1 above)
Comment (Be-ing) [reply res/controllers/common-controller-scripts.js]: ping

Comment (ywwg) [reply res/controllers/common-controller-scripts.js]: I changed this because it factor was undefined but rate was (which I think can happen in javascript if the variable passed in is undefined?), it would be passed through to spinback without being set.  Mostly it's a noop to remove code duplication.  Rate is checked in the existing code, below.

**Comment by ywwg** [comment]
> . From what I saw searching the web, the RCA inputs don't have phono preamps, but I do see a gain control that I suppose might provide enough gain for vinyl control to work.

I tried this out and I think it could work.  but vinyl pregain is currently broken, so I couldn't confirm.  I'm debugging the pregain to try to figure out what's broken

**Comment by ywwg** [comment]
ok I proposed a different PR to fix vinyl control pregain.  I was able to get a signal from the VCI, but it wasn't able to add enough gain to the phono signal to work with vinyl control (even after applying unlimited mixxx gain)

**Comment by ywwg** [comment]
(I confirmed that my regular Traktor Audio 10 still works with vinyl control :))

File: res/controllers/common-controller-scripts.js (hunk 1 above)
Comment (Be-ing) [reply res/controllers/common-controller-scripts.js]: Ah sorry, GitHub hid the code below so I missed that.

**Comment by ywwg** [comment]
thanks for the merge -- I'll look in to fixing the fx controls for the next version
